managed/CounterStrikeSharp.API/Core/Capabilities/Capabilities.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace CounterStrikeSharp.API.Core.Capabilities;
+
+public static class Capabilities
+{
+    public static void RegisterPluginCapability<T>(PluginCapability<T> capability, Func<T> supplier) where T : class
+    {
+        if (!PluginCapability<T>.Providers.ContainsKey(capability.Name))
+        {
+            PluginCapability<T>.Providers.Add(capability.Name, new());
+        }
+
+        PluginCapability<T>.Providers[capability.Name].Add(supplier);
+    }
+    
+    public static void RegisterPlayerCapability<T>(PlayerCapability<T> capability,
+        Func<CCSPlayerController, T> supplier) where T : class
+    {
+        if (!PlayerCapability<T>.Providers.ContainsKey(capability.Name))
+        {
+            PlayerCapability<T>.Providers.Add(capability.Name, new());
+        }
+
+        PlayerCapability<T>.Providers[capability.Name].Add(entity =>
+        {
+            if (entity is CCSPlayerController player)

```

Comment:
then I think this check is unnecessary

Reply:
Fixed https://github.com/roflmuffin/CounterStrikeSharp/pull/253/commits/5deaa5337a11823011a77751cb088cf3d9164941